### PR TITLE
[Ingest Manager] Improved verify experience

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

In this PR agent removes downloaded bits even if asc signature does not match (it was only the case for hash up until now)
Then fetch-verify is retried.

This helps mainly with sceanrio when artifacts are built without `.asc` files. So asc file downloaded from `snapshot` repository matches other build as self build binaries included in agent package. 
IN this case after initial failure, agent removes self build binaries and downloads snapshot from repository including `sha512` and `asc` files. 

BUT there's a bit more going on in this PR. there's added lock for update method in `emitter` because when dynamic input called Set and Config was Loaded it resulted in two concurrent processing of configuration which tried to setup and run own set of beats and they were fighting over `path.data` location, crashing... 

This problem sometimes manifested sometimes not, more probably during standalone scenario.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/core/plugin/process/app.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (a *Application) Name() string {
 
 // Started returns true if the application is started.
 func (a *Application) Started() bool {
-	return a.state.Status != state.Stopped
+	return a.state.Status != state.Stopped && a.state.Status != state.Crashed && a.state.Status != state.Failed
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/core/plugin/process/app.go:120]: Why this change?

I think `Started()` is trying to say if the application was started at all. If it is crashed or failed you don't want `Start()` to be called again, because that state is handled internally in the application abstraction.

File: x-pack/elastic-agent/pkg/core/plugin/process/start.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (a *Application) start(ctx context.Context, t app.Taggable, cfg map[string]
 	}()
 
 	// already started if not stopped or crashed
-	if a.state.Status != state.Stopped && a.state.Status != state.Crashed && a.state.Status != state.Failed {
+	if a.Started() {
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/core/plugin/process/start.go:42]: The internal `start()` is called to restart on a crash. We sure this change with the above change is not going to affect that behavior?

**Comment by blakerouse** [comment]
I think this change will fix #21120. As that issue has to do with concurrency of updates, that this added mutex will solve.

File: x-pack/elastic-agent/pkg/core/plugin/process/app.go (hunk 1 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/core/plugin/process/app.go:120]: yes in case verification fails, it says state to FAILING
when it passes in second round is tries to start the app but according to this check it is already running so it tries to configure it right away. 

i just unified this check with the one in `start.go` so start.go should be unaffected but `operation_start.Check` should start functioning as intended (not skipping operation)

File: x-pack/elastic-agent/pkg/core/plugin/process/start.go (hunk 2 above)
Comment (michalpristas) [reply x-pack/elastic-agent/pkg/core/plugin/process/start.go:42]: this should be ok we check it only 2 places here (which stays unaffected, it just looks more readable) 
and operation_start.Check which is called on config resolution

**Review by blakerouse — APPROVED** [review]
Thanks for the explanation. Looks good.
